md5.js

```diff
@@ -14,123 +14,122 @@
+  a = hh(a, b, c, d, k[1], 4, -1530992060);
+  d = hh(d, a, b, c, k[4], 11,  1272893353);
+  c = hh(c, d, a, b, k[7], 16, -155497632);
+  b = hh(b, c, d, a, k[10], 23, -1094730640);
+  a = hh(a, b, c, d, k[13], 4,  681279174);
+  d = hh(d, a, b, c, k[0], 11, -358537222);
+  c = hh(c, d, a, b, k[3], 16, -722521979);
+  b = hh(b, c, d, a, k[6], 23,  76029189);
+  a = hh(a, b, c, d, k[9], 4, -640364487);
+  d = hh(d, a, b, c, k[12], 11, -421815835);
+  c = hh(c, d, a, b, k[15], 16,  530742520);
+  b = hh(b, c, d, a, k[2], 23, -995338651);
+
+  a = ii(a, b, c, d, k[0], 6, -198630844);
+  d = ii(d, a, b, c, k[7], 10,  1126891415);
+  c = ii(c, d, a, b, k[14], 15, -1416354905);
+  b = ii(b, c, d, a, k[5], 21, -57434055);
+  a = ii(a, b, c, d, k[12], 6,  1700485571);
+  d = ii(d, a, b, c, k[3], 10, -1894986606);
+  c = ii(c, d, a, b, k[10], 15, -1051523);
+  b = ii(b, c, d, a, k[1], 21, -2054922799);
+  a = ii(a, b, c, d, k[8], 6,  1873313359);
+  d = ii(d, a, b, c, k[15], 10, -30611744);
+  c = ii(c, d, a, b, k[6], 15, -1560198380);
+  b = ii(b, c, d, a, k[13], 21,  1309151649);
+  a = ii(a, b, c, d, k[4], 6, -145523070);
+  d = ii(d, a, b, c, k[11], 10, -1120210379);
+  c = ii(c, d, a, b, k[2], 15,  718787259);
+  b = ii(b, c, d, a, k[9], 21, -343485551);
+
+  x[0] = add32(a, x[0]);
+  x[1] = add32(b, x[1]);
+  x[2] = add32(c, x[2]);
+  x[3] = add32(d, x[3]);
 }
 
 function cmn(q, a, b, x, s, t) {
-a = add32(add32(a, q), add32(x, t));
-return add32((a << s) | (a >>> (32 - s)), b);
+  a = add32(add32(a, q), add32(x, t));
+  return add32((a << s) | (a >>> (32 - s)), b);
 }
 
 function ff(a, b, c, d, x, s, t) {
-return cmn((b & c) | ((~b) & d), a, b, x, s, t);
+  return cmn((b & c) | ((~b) & d), a, b, x, s, t);
 }
 
 function gg(a, b, c, d, x, s, t) {
-return cmn((b & d) | (c & (~d)), a, b, x, s, t);
+  return cmn((b & d) | (c & (~d)), a, b, x, s, t);
 }
 
 function hh(a, b, c, d, x, s, t) {
-return cmn(b ^ c ^ d, a, b, x, s, t);
+  return cmn(b ^ c ^ d, a, b, x, s, t);
 }
 
 function ii(a, b, c, d, x, s, t) {
-return cmn(c ^ (b | (~d)), a, b, x, s, t);
+  return cmn(c ^ (b | (~d)), a, b, x, s, t);
 }
 
 function md51(s) {
-txt = '';
-var n = s.length,
-state = [1732584193, -271733879, -1732584194, 271733878], i;
-for (i=64; i<=s.length; i+=64) {
-md5cycle(state, md5blk(s.substring(i-64, i)));
-}
-s = s.substring(i-64);
-var tail = [0,0,0,0, 0,0,0,0, 0,0,0,0, 0,0,0,0];
-for (i=0; i<s.length; i++)
-tail[i>>2] |= s.charCodeAt(i) << ((i%4) << 3);
-tail[i>>2] |= 0x80 << ((i%4) << 3);
-if (i > 55) {
-md5cycle(state, tail);
-for (i=0; i<16; i++) tail[i] = 0;
-}
-tail[14] = n*8;
-md5cycle(state, tail);
-return state;
+  txt = '';
+  var n = s.length,
+  state = [1732584193, -271733879, -1732584194, 271733878], i;
+  for (i=64; i<=s.length; i+=64) {
+    md5cycle(state, md5blk(s.substring(i-64, i)));
+  }
+  s = s.substring(i-64);
+  var tail = [0,0,0,0, 0,0,0,0, 0,0,0,0, 0,0,0,0];
+  for (i=0; i<s.length; i++)
+  tail[i>>2] |= s.charCodeAt(i) << ((i%4) << 3);
```

Comment:
Expected '{' and instead saw 'tail'.<br>Unexpected use of '>>'.<br>Unexpected use of '|='.<br>Unexpected use of '<<'.